### ISSUE TITLE
CompletionService cannot be resolved in Blazor WASM

### ISSUE THREAD

**Issue by ConnerPhillis** [dotnet/roslyn#70404]
**Version: 4.8.0-2 Final** 

**Steps to Reproduce**:

Apologies in advance if I'm missing something, I'm still getting familiar with Roslyn. 

Running the code below in the Program.cs file in a Blazor WASM project will not resolve the C# CompletionService. In a standard .NET console app, the completion service is resolved.

Is this supported in WASM? I haven't been able to find the area where these completion providers are registered, if someone can point me there I'm happy to look into it myself.

code: 

```csharp
var adHocWorkspace = new AdhocWorkspace(MefHostServices.DefaultHost);
var projectInfo = ProjectInfo.Create(
	ProjectId.CreateNewId(),
	VersionStamp.Create(),
	"TemporaryProject",
	"TemporaryProject",
	LanguageNames.CSharp,
	metadataReferences: references,
	parseOptions: new CSharpParseOptions(
		kind: SourceCodeKind.Script,
		languageVersion: LanguageVersion.Latest));

var project = adHocWorkspace.AddProject(projectInfo);
var completionService = project.Services.GetService<CompletionService>();

var document = project.AddDocument("program.cs", "using System;\r\nvar g = Guid.NewGuid();");
var completionService2 = CompletionService.GetService(document);
```

**Expected Behavior**:

Completion Service is resolved

**Actual Behavior**:

null value returned by `CompletionService.GetService` and `project.Services.GetService<CompletionService>`

**Comment by CyrusNajmabadi** [comment]
> Is this supported in WASM?

We don't do anything to support this.  But we don't block it either.

Someone (ideally someone who is doing wasm development :D) would have to debug through to see what's failing.

**Comment by ConnerPhillis** [comment]
Hey Cyrus,

I've got no problem debugging through - but I also have zero clue where these services are being sourced from. I've worked through Project, ProjectInfo, and Solution, but haven't turned anything up yet. Any chance you have some pointers on where would be the best place to look for those things?

Also, am I correct in assuming that there is some sort of DI container going on behind the scenes for these services?

No problem if I'm asking too much, I'm fine to keep brute forcing it just hoping you can give me a shortcut :)

**Comment by ConnerPhillis** [comment]
Great! Thanks!

This is a project I've been doing in my free time, so I'll be working on it again later tonight. Hopefully I can turn something up :)

**Comment by ConnerPhillis** [comment]
@CyrusNajmabadi 

tl;dr; - stupid error on my end

Turns out I hadn't cleaned up nuget packages in my testing project so `Microsoft.CodeAnalysis.CSharp.Features` was not referenced in the service layer library that I had built. Adding a package reference for that package in my library resolved the issue, the `CSharpCompletionService` was able to be loaded.

While I was working on debugging this though I came up with a question: I'm guessing since the CSharpCompletionProvider is internal and not referenced by anything except test projects there's a high chance that it'll be trimmed. Since Blazor by default publishes trimmed I'm guessing that means it has to be disabled for this project?

**Comment by CyrusNajmabadi** [comment]
The type should be referenced.  Otherwise it wouldn't work in hosts like vs.  Unless you're using a trimmer that throws out types that mef would hook up.

**Comment by LHK-666** [comment]
Code:
public CodeCompletionService()
{
    workSpace = new AdhocWorkspace();
}

public Project CreateProject(string projectName)
{
    var projectId = ProjectId.CreateNewId(debugName: projectName);
    var projectInfo = ProjectInfo.Create(
        projectId,
        VersionStamp.Create(),
        projectName,
        projectName,
        LanguageNames.CSharp)
         .WithMetadataReferences(new[] { MetadataReference.CreateFromFile(typeof(object).Assembly.Location) });
    Project project = workSpace.AddProject(projectName,"C#");
    return project;
}
Hello, I would like to ask about the execution of var project=workSpace AddProject(projectInfo); When using the AddProject method in the code line. An error message appeared stating 'System. NotSupported Exception:' Language 'C #' is not supported. '
